File: LC/lc1299/lc1299.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
  let max = -1, max2 = -1;
  for (i = arr.length - 1; i >= 0; i--) {
      max2 = (max > arr[i]) ? max : arr[i];
      arr[i] = max;
      max = max2;
  }
  return arr;
};

File: LC/lc1299/lc1299.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        l = len(arr) - 1
        trs = arr[l]
        arr[l] = -1
        l -= 1
        while l >= 0:
            arr[l], trs = trs, max(arr[l], trs)
            l -= 1
        return arr